pkg/distributor/distributor.go

```diff
@@ -902,7 +899,7 @@ func (d *Distributor) forwardSamples(ctx context.Context, userID string, ts []mi
 
 // Push implements client.IngesterServer
 func (d *Distributor) Push(ctx context.Context, req *mimirpb.WriteRequest) (*mimirpb.WriteResponse, error) {
-	return d.PushWithMiddlewares(ctx, req, func() { mimirpb.ReuseSlice(req.Timeseries) })
+	return d.WrapPushWithMiddlewares(d.PushWithCleanup)(ctx, req, func() { mimirpb.ReuseSlice(req.Timeseries) })
```

Comment:
Wouldn't this now do the wrapping on every push? While previously the wrapping was only done once in `New()` and then the wrapped function was re-used on each push?